gnoland/website/static/js/wallet.js

```diff
@@ -0,0 +1,145 @@
+var remote = "http://gno.land:36657",
+    chainId = "testchain",
+    walletFn = {keplr: {}};
+
+walletFn.createVmCallMsg = function (sender, pkgPath, funcName, args) {
+	return  {
+		type: "/vm.m_call",
+		value: {
+			caller: sender,
+			send: "",
+			pkg_path: pkgPath,
+			func: funcName,
+			args: args,
+		}
+	};
+};
+
+walletFn.createSignDoc = function(account, msg, chainId, gas) {
+	return {
+	  msgs: [msg],
+	  fee: { amount: [{
+		amount: "1",
+		denom: "gnot"
+	  }], gas: gas },
+	  chain_id: chainId,
+	  memo: "",
+	  account_number: account.account_number,
+	  sequence: account.sequence,
+	};
+};
+
+walletFn.keplr.signAndBroadcast = function(sender, msg) {
+	return window.keplr.experimentalSuggestChain(walletFn.geTestnetKeplrConfig())
+	.then(function () {
+		return window.keplr.enable(chainId);
+	})
+	.then(function () {
+		return walletFn.getAccount(sender);
+	})
+	.then(function(account) {
+		const signDoc = walletFn.createSignDoc(account, msg, chainId, "2000000");
+		return window.keplr.signAmino(chainId, sender, signDoc, {
+			// use app fee (1gnot fixed fee)
+			preferNoSetFee: true, 
+		});
+	})
+	.then(function (signature) {
+		const tx = walletFn.makeStdTx(signature.signed, signature.signature);
+		return walletFn.broadcastTx(tx);
+	});
+};
+
+walletFn.geTestnetKeplrConfig = function() {
+	const addressPrefix = "g";
+	const gnoToken = {
+		coinDenom: "GNOT",
+		coinMinimalDenom: "gnot",
+		coinDecimals: 6,
```

Comment:
TIL :)

where is this number coming from?

Reply:
from ATOM, I'm not sure if Jae wants to change that.